lib/linter/linter.js

```diff
@@ -623,6 +623,56 @@ function analyzeScope(ast, parserOptions, visitorKeys) {
     });
 }
 
+/**
+ * Checks whether a given AST belongs to a generated file
+ * @param {ExtendedProgam} ast The AST to check
+ * @returns {boolean} true if the AST seems to be from a generated file
+ */
+function isGeneratedAST(ast) {
+    const TOKEN_DENSITY_THRESHOLD = 150;
+    const IDENTIFIER_THRESHOLD = 40;
+    const LOC_THRESHOLD = 20000;
+
+    const { loc } = ast;
+
+    if (!loc) {
+        throw new Error("Location information not found in the AST");
+    }
+
+    const { tokens } = ast;
+
+    if (!tokens) {
+        throw new Error("Token information not found in the AST");
+    }
+
+    const linesOfCode = (loc.end.line - loc.start.line) || 1;
+    const tokenDensity = tokens.length / linesOfCode;
+
+    let identifierCount = 0;
+    let singleCharIdentifierCount = 0;
+
+    for (const token of tokens) {
+        if (token.type === "Identifier") {
+            identifierCount++;
+            if (token.value.length === 1) {
+                singleCharIdentifierCount++;
+            }
+        }
+    }
+
+    const singleCharIdentifierPercent = (singleCharIdentifierCount / identifierCount) * 100;
+    const isGenerated =
+        !!(tokenDensity > TOKEN_DENSITY_THRESHOLD ||
```

Comment:
can we use `Boolean(foo)` instead of `!!foo`?

Reply:
Actually, the Airbnb style guide recommends `!!` as the best practice https://github.com/airbnb/javascript#coercion--booleans and the ESLint autofix also follows this.

I prefer doing this but am open to change if moving to `Boolean` is more beneficial. LMK.